apps/web/pages/auth/verify.tsx

```diff
@@ -158,7 +164,7 @@ export default function Verify() {
                 e.preventDefault();
                 setSecondsLeft(30);
                 // Update query params with t:timestamp, shallow: true doesn't re-render the page
-                const _searchParams = new URLSearchParams(searchParams);
+                const _searchParams = new URLSearchParams(searchParams.toString());
```

Comment:
it was type erroring here but you can to searchParams.toString and the new URLSearchParams should be able to pickup url encoded string.